app/src/models/repository.ts

```diff
@@ -72,11 +76,27 @@ export class Repository {
       this.workflowPreferences.forkContributionTarget,
       this.isTutorialRepository
     )
+    this.fetchUrl()
   }
 
   public get path(): string {
     return this.mainWorkTree.path
   }
+
+  public get url(): string | null {
+    return this._url
+  }
+
+  private fetchUrl(): void {
+    getRemotes(this).then(remotes => {
+      const defaultRemote = findDefaultRemote(remotes)
+      if (defaultRemote) {
+        this._url = defaultRemote.url
+      } else {
+        this._url = null
+      }
+    })
+  }
```

Comment:
Preferably, we would not want to add `git` calls for every repository when it's only use case is non-github repositories and only when a user wants to open the repository in the browser.  Could we look creating a `getNonGitHubUrl` method similar to `getGitHubHtmlUrl` used in the now `app.tsx > viewInBrowser` method. So it would look something like `const url = getGitHubHtmlUrl(repository) ?? getNonGitHubHtmlURL(repository)`

We could likely update the `getRemotes` call to only accept a repositoryPath since that is all that is used from the `Repository` object that is sent in, and we can make `getNonGitHubHtmlURL` utilize `memoizeOne` such that if the same repo path is sent in, it will not have to do the `git` call again.

Reply:
This maybe best as an exportable method in the `repository.ts` so it can be reused in the menu-update.ts

```
menuStateBuilder.setEnabled(
      'view-repository-on-github',
      isHostedOnGitHub || hasOriginUrl
    )
 ```